Crash in Coptic calendar

#### Software and hardware versions 
Collect v1.12.0-beta.0

#### Problem description
```
Caused by: java.lang.NullPointerException: 
  at org.joda.time.LocalDateTime.getYear (LocalDateTime.java:1562)
  at org.joda.time.LocalDateTime.toDateTime (LocalDateTime.java:750)
  at org.joda.time.DateTimeZone.isLocalDateTimeGap (DateTimeZone.java:1166)
  at org.odk.collect.android.utilities.DateTimeUtils.skipDaylightSavingGapIfExists (DateTimeUtils.java:88)
  at org.odk.collect.android.fragments.dialogs.CopticDatePickerDialog.setUpDatePicker (CopticDatePickerDialog.java:60)
```

#### Steps to reproduce the problem

#### Expected behavior

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

--- a/collect_app/src/main/java/org/odk/collect/android/utilities/DateTimeUtils.java
+++ b/collect_app/src/main/java/org/odk/collect/android/utilities/DateTimeUtils.java
@@ -85,8 +85,11 @@ public class DateTimeUtils {
 
     public static LocalDateTime skipDaylightSavingGapIfExists(LocalDateTime date) {
         final DateTimeZone dtz = DateTimeZone.getDefault();
-        while (dtz.isLocalDateTimeGap(date)) {
-            date = date.plusMinutes(1);
+
+        if (dtz != null) {
+            while (dtz.isLocalDateTimeGap(date)) {
+                date = date.plusMinutes(1);
+            }
         }
         return date;
     }

